### ISSUE TITLE
question about fig2 in the paper.

### ISSUE THREAD

**Issue by dingangui** [ToTheBeginning/PuLID#27]
Hello, thanks for your incredible work!

In the 'Accurate ID Loss' section in the bottom right corner of Figure 2 of the paper, there are two generated images both denoted as 'predict x_0'. Are both of these images produced by the Lighting T2I? I guess they represent T2I w/ ID and T2I w/o ID, respectively. However, upon closer inspection, it appears that the IDs of both images are well-preserved, which contradicts my speculation. What are these two images' actual meanings and why do you connect them with a vertical line? 
<img width="286" alt="image" src="https://github.com/ToTheBeginning/PuLID/assets/30258241/ee0774a1-fef3-4bd0-a445-1fd541fc174a">

**Comment by guozinan126** [comment]
When calculating the ID loss, the two images involved in the calculation are both generated by the Lightning T2I training branch, and both images are generated under the T2I w/ ID setting. Additionally, only when calculating the Alignment loss, the contrastive pair is composed of T2I w/ ID and T2I w/o ID.

**Comment by Luh1124** [comment]
> When calculating the ID loss, the two images involved in the calculation are both generated by the Lightning T2I training branch, and both images are generated under the T2I w/ ID setting. Additionally, only when calculating the Alignment loss, the contrastive pair is composed of T2I w/ ID and T2I w/o ID.

Is there no id loss calculated with the given identity images here? Is it only done on the generated images? Also, I would like to ask, how is it ensured that the ID loss done by ArcFace is differentiable for gradient backpropagation?
